homeassistant/components/sensor/__init__.py

```diff
@@ -77,15 +83,62 @@ async def async_unload_entry(hass, entry):
     return await hass.data[DOMAIN].async_unload_entry(entry)
 
 
-class SensorEntity(MeasurableUnitEntity):
+class SensorEntity(Entity):
     """Base class for sensor entities."""
 
     @property
-    def native_value(self):
+    def native_value(self) -> str | None:
         """Return the value reported by the sensor."""
         return None
 
     @property
-    def native_unit_of_measurement(self):
+    def native_unit_of_measurement(self) -> str | None:
         """Return the unit of measurement of the sensor, if any."""
         return None
+
+    @property
+    def unit_of_measurement(self) -> str | None:
+        """Return the unit of measurement of the entity, after unit conversion."""
+        native_unit_of_measurement = self.native_unit_of_measurement
+
+        if native_unit_of_measurement in (TEMP_CELSIUS, TEMP_FAHRENHEIT):
+            # Suppress AttributeError to handle tests testing directly on entity objects
+            with suppress(ValueError, AttributeError):
+                return self.hass.config.units.temperature_unit
+
+        return native_unit_of_measurement
+
+    @property
+    def state(self) -> Any:
+        """Return the state of the sensor and perform unit conversions, if needed."""
+
+        unit_of_measurement = self.native_unit_of_measurement
+        value = self.native_value
+
+        # Convert temperature if we detect one
+        # Suppress ValueError (Could not convert sensor_value to float)
+        # Suppress AttributeError (Handle tests testing directly on entity objects)
+        with suppress(ValueError, AttributeError):
```

Comment:
Can we decrease the scope of this block to just the line where it will raise and return early in that case?

Reply:
Improved in dc3cad1